spec/queries/get_grouped_content_and_links_spec.rb

```diff
@@ -0,0 +1,211 @@
+require "rails_helper"
+
+RSpec.describe Queries::GetGroupedContentAndLinks do
+  let(:ordered_content_ids) do
+    %w(
+      5f612a9f-7d3f-4b4c-a35e-ebd0b1e79019
+      fd161210-3e12-41e7-8e5e-c8cef607a95f
+    )
+  end
+
+  def new_content_item
+    SecureRandom.uuid
+  end
+
+  describe "#call" do
+    context "when no results exist" do
+      it "returns an empty array" do
+        expect(subject.call).to be_empty
+      end
+    end
+
+    def create_content_items(quantity)
+      quantity.times do
+        FactoryGirl.create(
+          :content_item,
+          base_path: "/random-base-path-#{Random.rand}",
+        )
+      end
+    end
+
+    context "when no pagination is specified" do
+      it "returns page with default page size" do
+        create_content_items(described_class::PAGE_SIZE + 1)
+        expect(subject.call.size).to eql(described_class::PAGE_SIZE)
+      end
+    end
+
+    context "when retrieving the next page" do
+      it "returns items after last seen" do
+        item = FactoryGirl.create(
+          :content_item,
+          base_path: '/random',
+          content_id: ordered_content_ids.first
+        )
+
+        item2 = FactoryGirl.create(
+          :content_item,
+          content_id: ordered_content_ids.last
+        )
+
+        results = described_class.new(last_seen_content_id: item.content_id).call
+
+        expect(results.size).to eq(1)
+        expect(results[0]["content_id"]).to eql(item2.content_id)
+      end
+    end
+
+    context "when there are documents" do
+      before do
+        @content_item = FactoryGirl.create(
+          :content_item,
+          content_id: ordered_content_ids.first,
+          base_path: "/capital-gains-tax",
+          state: "published"
+        )
+      end
+
+      context "with no links" do
+        it "returns the content item with empty links" do
```

Comment:
This may be testing more than the description - perhaps you could just assert `expect(result["links"]).to eq([])` here, since the presence of the content is already tested in the request tests? Or perhaps split it up somehow?